stellar-contract-env-common/src/static_val.rs

```diff
@@ -0,0 +1,15 @@
+use stellar_xdr::ScStatic;
+
+use crate::{TagStatic, TaggedVal};
+
+pub type Static = TaggedVal<TagStatic>;
+impl Static {
+    // NB: we don't provide a "get_type" to avoid casting a bad bit-pattern into
+    // an ScStatic. Instead we provide an "is_type" to check any specific
+    // bit-pattern.
+
+    #[inline(always)]
+    pub const fn is_type(&self, ty: ScStatic) -> bool {
+        self.0.has_minor(ty as u32)
+    }
```

Comment:
The is_type interface is convenient that it avoids the bad cast, but it results in suboptimal code everywhere else.  What if instead of defining a Static that is simply the TaggedVal type, we make it its own type, and make it an enum. So you convert from the `TaggedVal<TagStatic>` via a `TryFrom` impl into a `Static`. The conversion fails for an unknown/unrecognized type. From there code gets all the benefits of Rust enums, like exhaustive checks, etc.

Reply:
Wouldn't the `TryFrom` still need to use `is_type` with what you're proposing to go from `TaggedVal<TagStatic>` to `Static`?

Reply:
Ah yes, it would need that function also.